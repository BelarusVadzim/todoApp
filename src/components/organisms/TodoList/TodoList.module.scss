@import "src/variables.scss";

.todoList {
  border-left: 1rem transparent solid;
  border-right: 1rem transparent solid;
  margin: auto;
  height: $todo-list-height;
  max-width: $todo-list-max-width;
  min-width: $todo-list-min-width;
  position: relative;
  top: $todo-list-top;
}

.todoTtitle {
  height: $todoTtitle-height;
  font-size: $todoTtitle-font-size;
}

.noteEditor {
  composes: container from "~/src/styles/container.scss";
  height: $noteEditor-height;
  margin: 2rem 0;
}

.listContainer {
  composes: container from "~/src/styles/container.scss";
  position: relative;
}
