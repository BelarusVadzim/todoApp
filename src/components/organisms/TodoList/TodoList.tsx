import style from './TodoList.module.scss';
import {
  NoteEditor,
  ScrolableSection,
  TodoListFooter,
  TodoItemsListMenu,
  Spinner,
} from 'components/molecules';
import { useTodoStateService } from 'hooks';

const TodoList: React.FC = () => {
  const { pending } = useTodoStateService();

  return (
    <div className={style.todoList}>
      <h1 className={style.todoTtitle}>TODO</h1>
      <NoteEditor className={style.noteEditor} />
      <div className={style.listContainer}>
        <ScrolableSection />
        <TodoItemsListMenu />
      </div>
      <Spinner visible={pending} />
      <TodoListFooter />
    </div>
  );
};

export default TodoList;
