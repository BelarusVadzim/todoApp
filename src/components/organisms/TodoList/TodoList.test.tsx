import { render } from '@testing-library/react';
import TodoList from './TodoList';

jest.mock('components/molecules', () => ({
  ScrolableSection: jest.fn(() => <div>ScrolableSection</div>),
  TodoListFooter: jest.fn(() => <div>TodoListFooter</div>),
  TodoItemsListMenu: jest.fn(() => <div>TodoItemsListMenu</div>),
  NoteEditor: jest.fn(({ className }) => {
    return <div className={className} />;
  }),
  TodoTitle: jest.fn(({ className, children }) => {
    return <div className={className}>{children}</div>;
  }),
}));

afterEach(() => {
  jest.clearAllMocks();
});

describe('<TodoList />', () => {
  it('should render properly', () => {
    const component = render(<TodoList />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
