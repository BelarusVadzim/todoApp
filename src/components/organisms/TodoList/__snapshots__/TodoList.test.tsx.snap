// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="todoList"
  >
    <h1
      class="todoTtitle"
    >
      TODO
    </h1>
    <div
      class="noteEditor"
    />
    <div
      class="listContainer"
    >
      <div>
        ScrolableSection
      </div>
      <div>
        TodoItemsListMenu
      </div>
    </div>
    <div>
      TodoListFooter
    </div>
  </div>
</DocumentFragment>
`;
